mopidy/core/actor.py

```diff
@@ -136,6 +138,106 @@ def tags_changed(self, tags):
             self.playback._stream_title = title
             CoreListener.send('stream_title_changed', title=title)
 
+    def on_start(self):
+        logger.debug("core on_start")
+        try:
+            coverage = []
+            if self._config and 'restore_state' in self._config['core']:
+                amount = self._config['core']['restore_state']
+                if not amount or 'off' == amount:
+                    pass
+                elif 'volume' == amount:
+                    coverage = ['volume']
+                elif 'load' == amount:
+                    coverage = ['tracklist', 'mode', 'volume', 'history']
+                elif 'last' == amount:
+                    coverage = ['tracklist', 'mode', 'play-last', 'volume',
+                                'history']
+                elif 'play' == amount:
+                    coverage = ['tracklist', 'mode', 'play-always', 'volume',
+                                'history']
+                else:
+                    logger.warn('Unknown value for config '
+                                'core.restore_state: %s', amount)
+            if len(coverage):
+                self.load_state('persistent', coverage)
+        except Exception as e:
+            logger.warn('Unexpected error: %s', str(e))
+        pykka.ThreadingActor.on_start(self)
+
+    def on_stop(self):
+        logger.debug("core on_stop")
+        try:
+            if self._config and 'restore_state' in self._config['core']:
+                amount = self._config['core']['restore_state']
+                if amount and 'off' != amount:
+                    self.save_state('persistent')
+        except Exception as e:
+            logger.warn('on_stop: Unexpected error: %s', str(e))
+        pykka.ThreadingActor.on_stop(self)
+
+    def save_state(self, name):
+        """
+        Save current state to disk.
+
+        :param name: a name (for later use with :meth:`load_state`)
```

Comment:
My thinking was that we start out without a name to keep the API and code implementing it simpler. Then if there is a need for storing multiple states we can always expand with an optional name argument.

So basically just trying to stick to having minimal additions to the API at a time as it's easy to add and expand, but really a lot more work to remove things. So I err on the side of adding less and having a plan for how to add more later if needed.